Userland/Libraries/LibCompress/Lzma.cpp

```diff
@@ -520,22 +520,13 @@ ErrorOr<Bytes> LzmaDecompressor::read_some(Bytes bytes)
             if (m_options.uncompressed_size.has_value() && m_options.uncompressed_size.value() < m_total_decoded_bytes + real_length)
                 return Error::from_string_literal("Tried to copy match beyond expected uncompressed file size");
 
-            while (real_length > 0) {
-                if (m_dictionary->empty_space() == 0) {
-                    m_leftover_match_length = real_length;
-                    break;
-                }
+            auto copied_length = TRY(m_dictionary->copy_from_seekback(current_repetition_offset(), real_length));
```

Comment:
Glad this method is useful outside of DEFLATE!

Reply:
I certainly should think about adding common helpers more often. In this case we both get shared improvements _and_ the advantage of being able to rely on internals, which essentially skips one of two copy operations. :^)